pkgs/development/python-modules/sphinxcontrib-ditaa/default.nix

```
@@ -0,0 +1,40 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, unittestCheckHook
+, mock
+, sphinx-testing
+, sphinx
+, ditaa
+}:
+
+buildPythonPackage rec {
+  pname = "sphinxcontrib-ditaa";
+  version = "1.0.2";
+  format = "setuptools";
```

```suggestion
  pyproject = true;
```
Please use `pyproject = true` instead to prevent a legacy build.
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#buildpythonpackage-parameters-buildpythonpackage-parameters

---

Looks like although pyproject=true tries to fallback to setuptools,  it's not entirely turnkey.    Get this trace:

```
error: builder for '/nix/store/jlcx6d5n43l1a4vwvb35ixnni76mc9pq-python3.11-sphinxcontrib-ditaa-1.0.2.drv' failed with exit code 1;
       last 10 log lines:
       >   File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
       >   File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
       >   File "<frozen importlib._bootstrap>", line 1126, in _find_and_load_unlocked
       >   File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
       >   File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
       >   File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
       >   File "<frozen importlib._bootstrap>", line 1140, in _find_and_load_unlocked
       > ModuleNotFoundError: No module named 'setuptools'
       >
       > ERROR Backend 'setuptools.build_meta:__legacy__' is not available.
       For full logs, run 'nix-store -l /nix/store/jlcx6d5n43l1a4vwvb35ixnni76mc9pq-python3.11-sphinxcontrib-ditaa-1.0.2.drv'.
error: 1 dependencies of derivation '/nix/store/caxg5qf24skhkik07k845mk91iy16j82-nix-shell-env.drv' failed to build
```

Trying with explicit setuptools dependency